src/unix/apple/disk.rs

```diff
@@ -91,7 +91,23 @@ impl crate::DisksInner {
     }
 
     pub(crate) fn refresh_list(&mut self) {
+        struct AutoreleasePool {
+            ctx: *mut c_void,
+        }
+
+        impl Drop for AutoreleasePool {
+            fn drop(&mut self) {
+                // SAFETY: We have not manipulated `pool_ctx` since it was received from a corresponding
+                // pool push call.
+                unsafe { ffi::objc_autoreleasePoolPop(self.ctx) }
+            }
+        }
+
         unsafe {
+            // SAFETY: Creating a new pool is safe in any context. They can be arbitrarily nested
+            // as long as pool objects are not used in deeper layers, but we only have one and don't
+            // allow it to leave this scope.
+            let _refresh_pool = AutoreleasePool { ctx: ffi::objc_autoreleasePoolPush() };
```

Comment:
In this case, I think either make both C calls here or you can add a `AutoreleasePool::new` method which will call `objc_autoreleasePoolPush`. This in-between looks a bit strange to me.

Reply:
I moved this to be all Rust calls. I did need to switch in a closure-based approach though since release pools are stack-frame sensitive, so a `new` function that is not guaranteed to be inlined might be problematic. On the upside, the code is more generic if somewhere else needs this in the future again.

Reply:
Just I really like this approach. I just think that having a struct is useless at this point.